### DISCUSSION TITLE
Map becomes too slow when rendering more than 1000 geoJSON features using GoogleMapsOverlay with stroked enabled.

### DISCUSSION THREAD

**Discussion by aravindr0698** [visgl/deck.gl#9227] [Q&A]
When passing `stroked:true` to create borders around the features while creating an geojsonoverlay, map becomes too slow. Passing stroked as false resolves the issue, but I need borders around my features. I need to render around 10K features on map. Is there any workaround or am I missing something?

``new GeoJsonLayer({
        id: 'airports2',
        data: 'https://raw.githubusercontent.com/OpenDataDE/State-zip-code-GeoJSON/refs/heads/master/ak_alaska_zip_codes_geo.min.json',
        // Styles
        stroked: true,
        filled: true,
        pointRadiusMinPixels: 2,
        lineWidthScale: 20,
        lineWidthMinPixels: 1,
        getFillColor: [160, 160, 180, 200],
        pointRadiusScale: 2000,
        getFillColor: [200, 0, 80, 180],
      }),``
                
 This is what I'm trying to achieve 
<img width="1512" alt="Screenshot 2024-10-26 at 11 11 29 AM" src="https://github.com/user-attachments/assets/644e310b-7921-49c2-b30f-9737a3ccb654">

**Comment by TextTL** [comment]
What's your GPU spec? I render 100K features on 1650 without issues.

**Reply by aravindr0698** [reply]
I'm using an 14-core GPU in an Apple MacBook M1 Pro 2021 model which I believe has more or less the same capacity as yours.

**Comment by jfuehner** [comment]
With attempting to render that many objects I would suggest looking at the deck documentation on using the [binary ](https://deck.gl/docs/developer-guide/performance#use-binary-data) format or async iterables. I suspect a lot of processing is being done on the CPU prior to deck elevating to the GPU for rendering. Using the deck binary format would allow you to in many cases bypass the CPU.
